diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 26e5a951..98a06e31 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -138,6 +138,9 @@ set_target_properties( exiv2lib PROPERTIES
     COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
 )
 
+# TIFF reader is trying to read XMP even when it is disabled and generates warnings
+add_compile_definitions(SUPPRESS_WARNINGS)
+
 if (MSVC)
     set_target_properties(exiv2lib PROPERTIES LINK_FLAGS "/ignore:4099")
 endif()
