diff --git a/CMakeLists.txt b/CMakeLists.txt
index 07af451e..0a6169a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,8 @@ set( PACKAGE_BUGREPORT                "http://github.com/exiv2/exiv2" )
 set( PACKAGE_URL                      "https://exiv2.org")
 set( PROJECT_DESCRIPTION              "Exif/IPTC/Xmp C++ metadata library and tools plus ICC Profiles, Previews and more.")
 
+set(CMAKE_CXX_STANDARD 17)
+
 if ( EXIV2_ENABLE_EXTERNAL_XMP )
     set(EXIV2_ENABLE_XMP OFF)
 endif()
diff --git a/cmake/compilerFlags.cmake b/cmake/compilerFlags.cmake
index 68fd8f0d..f0f95de1 100644
--- a/cmake/compilerFlags.cmake
+++ b/cmake/compilerFlags.cmake
@@ -150,4 +150,29 @@ if(MSVC)
         add_compile_options("/Zc:__cplusplus")
     endif()
 
+    #
+    # We cannot use RelWithDebInfo because our Nuget dependencies have
+    # only Debug and Release configurations. Having debug symbols has
+    # no downside and helps a lot when troubleshooting issues in released
+    # binaries, so we add /Zi to the release configuration if it's not
+    # there already.
+    #
+    string(FIND ${CMAKE_CXX_FLAGS_RELEASE} "/Zi" HAS_MSVC_Zi)
+
+    if(HAS_MSVC_Zi EQUAL -1)
+        string(APPEND CMAKE_CXX_FLAGS_RELEASE " /Zi")
+    endif()
+
+    #
+    # This is an attempt to work around a bug in VC++ toolsets 14.33/14.34,
+    # which incorrectly optimize std::string construction, leaving internal
+    # string representation without a null terminator.
+    #
+    string(REGEX MATCH "/O[12]" HAS_MSVC_Ox ${CMAKE_CXX_FLAGS_RELEASE})
+
+    if(HAS_MSVC_Ox STRGREATER "")
+        string(REGEX REPLACE "/O[12]" "/Od" CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
+    else()
+        string(APPEND CMAKE_CXX_FLAGS_RELEASE " /Od")
+    endif()
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 26e5a951..98a06e31 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -138,6 +138,9 @@ set_target_properties( exiv2lib PROPERTIES
     COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
 )
 
+# TIFF reader is trying to read XMP even when it is disabled and generates warnings
+add_compile_definitions(SUPPRESS_WARNINGS)
+
 if (MSVC)
     set_target_properties(exiv2lib PROPERTIES LINK_FLAGS "/ignore:4099")
 endif()
