--- "exiv2-0.28.3_base\\build\\src\\exiv2lib_int.vcxproj"	2024-09-24 19:44:32.168188900 -0400
+++ "exiv2-0.28.3\\build\\src\\exiv2lib_int.vcxproj"	2024-09-24 21:49:06.260785500 -0400
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="17.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.props" Condition="Exists('..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.props')" />
   <PropertyGroup>
     <PreferredToolArchitecture>x64</PreferredToolArchitecture>
   </PropertyGroup>
@@ -34,6 +35,7 @@
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <CharacterSet>MultiByte</CharacterSet>
     <PlatformToolset>v143</PlatformToolset>
+    <UseDebugLibraries>true</UseDebugLibraries>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
@@ -341,7 +343,18 @@
       <CopyToOutputDirectory>Never</CopyToOutputDirectory>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <None Include="packages.config" />
+  </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
+    <Import Project="..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.targets" Condition="Exists('..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.targets')" />
   </ImportGroup>
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.props'))" />
+    <Error Condition="!Exists('..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\StoneSteps.zLib.VS2022.Static.1.3.1.1\build\native\StoneSteps.zLib.VS2022.Static.targets'))" />
+  </Target>
 </Project>
\ No newline at end of file
