diff --git a/CMakeLists.txt b/CMakeLists.txt
index 07af451e..b3b85bac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,10 +13,10 @@ project(exiv2          # use TWEAK to categorize the build
 include(cmake/mainSetup.cmake  REQUIRED)
 
 # options and their default values
-option( BUILD_SHARED_LIBS             "Build exiv2lib as a shared library"                    ON  )
-option( EXIV2_ENABLE_XMP              "Build with XMP metadata support"                       ON  )
+option( BUILD_SHARED_LIBS             "Build exiv2lib as a shared library"                    OFF )
+option( EXIV2_ENABLE_XMP              "Build with XMP metadata support"                       OFF )
 option( EXIV2_ENABLE_EXTERNAL_XMP     "Use external version of XMP"                           OFF )
-option( EXIV2_ENABLE_PNG              "Build with png support (requires libz)"                ON  )
+option( EXIV2_ENABLE_PNG              "Build with png support (requires libz)"                OFF )
 option( EXIV2_ENABLE_NLS              "Build native language support (requires gettext)"      OFF )
 option( EXIV2_ENABLE_PRINTUCS2        "Build with Printucs2"                                  ON  )
 option( EXIV2_ENABLE_LENSDATA         "Build including lens data"                             ON  )
@@ -28,8 +28,8 @@ option( EXIV2_ENABLE_CURL             "USE Libcurl for HttpIo (WEBREADY)"
 option( EXIV2_ENABLE_SSH              "USE Libssh for SshIo (WEBREADY)"                       OFF )
 option( EXIV2_ENABLE_BMFF             "Build with BMFF support"                               OFF )
 
-option( EXIV2_BUILD_SAMPLES           "Build sample applications"                             ON  )
-option( EXIV2_BUILD_EXIV2_COMMAND     "Build exiv2 command-line executable"                   ON  )
+option( EXIV2_BUILD_SAMPLES           "Build sample applications"                             OFF )
+option( EXIV2_BUILD_EXIV2_COMMAND     "Build exiv2 command-line executable"                   OFF )
 option( EXIV2_BUILD_UNIT_TESTS        "Build unit tests"                                      OFF )
 option( EXIV2_BUILD_FUZZ_TESTS        "Build fuzz tests (libFuzzer)"                          OFF )
 option( EXIV2_BUILD_DOC               "Add 'doc' target to generate documentation"            OFF )
